realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -723,8 +723,14 @@ private void emitCreateExpectedObjectSchemaInfo(JavaWriter writer) throws IOExce
                 "createExpectedObjectSchemaInfo", // Method name
                 EnumSet.of(Modifier.PRIVATE, Modifier.STATIC)); // Modifiers
 
+        // Guess capacity for Arrays used by OsObjectSchemaInfo.
+        // Used to prevent array resizing at runtime
+        int persistedFields = metadata.getFields().size();
+        int computedFields = metadata.getBacklinkFields().size();
+
         writer.emitStatement(
-                "OsObjectSchemaInfo.Builder builder = new OsObjectSchemaInfo.Builder(\"%s\")", this.simpleClassName);
+                "OsObjectSchemaInfo.Builder builder = new OsObjectSchemaInfo.Builder(\"%s\", %s, %s)",
+                this.simpleClassName, persistedFields, computedFields);
```

Comment:
I have this weird idea that you could probably just pass in the `metadata` as is, that way you wouldn't need to directly pass `int`s as parameters to the constructor.

Reply:
Not entirely sure what you mean? `OsObjectSchemaInfo.Builder` contains two internal arrays. You mean that those should be constructed outside and then parsed in?

Reply:
Actually never mind, there's a lot more to do with these sizes than I thought (JNI)